src/ChangeFeedIterator.ts

```diff
@@ -101,14 +129,21 @@ export class ChangeFeedIterator<T> {
       feedOptions.partitionKey = this.changeFeedOptions.partitionKey as any; // TODO: our partition key is too restrictive on the main object
     }
 
-    return this.clientContext.queryFeed<T>(
+    const response: Response<Array<T & Resource>> = await (this.clientContext.queryFeed<T>(
       this.resourceLink,
       ResourceType.item,
       this.resourceId,
       result => (result ? result.Documents : []),
       undefined,
       feedOptions,
       this.changeFeedOptions.partitionKeyRangeId
-    ) as Promise<any>; // TODO: some funky issues with query feed. Probably need to change it up.
+    ) as Promise<any>); // TODO: some funky issues with query feed. Probably need to change it up.
+
+    return new ChangeFeedResponse(
+      response.result,
```

Comment:
any reason these params need to be split up? Could just pass `response` and keep all this logic inside  `ChangeFeedResponse

Reply:
Call it personal preference. I like to make it clear what an object requires. I like being able to pass a null result or something, in the future, without having to mutate the response object.

Also, I like using the `public` syntax on objects, as well, so I don't need to unpack in the constructor.

Thirdly, I don't like the current `Response` object, and I want to get rid of it in v3, so it's one less object I've gotta touch.

Reply:
Got it. You've convinced me.